io-engine/src/bdev/ftl.rs

```diff
@@ -0,0 +1,258 @@
+//!
+//! The ftl bdev creates a CSAL-FTL disk.
+use std::{
+    collections::HashMap,
+    convert::TryFrom,
+    fmt::{Debug, Formatter},
+};
+
+use core::ffi::c_void;
+
+use async_trait::async_trait;
+use futures::channel::oneshot;
+use log::info;
+use nix::errno::Errno;
+use snafu::{OptionExt, ResultExt};
+use std::mem;
+use url::Url;
+
+use spdk_rs::ffihelper::errno_result_from_i32;
+use spdk_rs::{
+    libspdk::{
+        bdev_ftl_create_bdev, bdev_ftl_delete_bdev, ftl_bdev_info,
+        spdk_ftl_conf, spdk_ftl_get_default_conf,
+    },
+    UntypedBdev,
+};
+
+use crate::{
+    bdev::{dev::reject_unknown_parameters, util::uri, CreateDestroy, GetName},
+    bdev_api::{self, bdev_create, bdev_destroy, BdevError},
+    core::VerboseError,
+    ffihelper::{cb_arg, done_errno_cb, ErrnoResult},
+};
+
+pub struct Ftl {
+    /// the name of the ftl-bdev we created
```

Comment:
Try to start with a `C`apital and end with a `.` - makes the auto-generated rust docs look better IIRC

```suggestion
    /// The name of the ftl-bdev we created.
```